spec/support/test_contamination.rb

```diff
@@ -0,0 +1,19 @@
+module Spec
+  module Support
+    module TestContamination
+      def self.setup
+        TOPLEVEL_BINDING.eval('self').method(:include).owner.prepend(patched_include)
+      end
+
+      def self.patched_include
+        Module.new do
+          def include(included)
+            exceptions = ["ActionView::Helpers::NumberHelper", "UiConstants"] # TODO: Why is UiConstants included in `main`??? :(
+            return super if included.to_s.in?(exceptions)
+            raise RuntimeError, "#{included} included in #{self}", caller
```

Comment:
@bdunne sorry, can you make this more clear?  We'll not know what it's complaining about.